### DISCUSSION TITLE
SQLite version in SQLpage

### DISCUSSION THREAD

**Discussion by lyderic** [sqlpage/SQLPage#434] [Q&A] [answered]
Hello,

My version of SQLite is:

```bash
$ pacman -Q sqlite 
sqlite 3.46.0-1
```

However, if I run this in SQLpage (version 0.23.0):

```sql
SELECT 'debug' AS component, sqlite_version();
```

I get:

```json
{"component":"debug","sqlite_version()":"3.45.0"}
```

I understand that the SQLpage binary contains its own SQLite binary embedded into it, which explains the difference in versions between what's installed on my system and what's showing up on a web page produced with SQLpage.

That said, is there a reason why SQLpage doesn't update its embedded SQLite along with the SQLite project?

For information, [SQLite 3.46.0](https://www.sqlite.org/releaselog/3_46_0.html) was released on 23 May 2024 whereas SQLpage 0.23.0 (most recent version when I am writing these lines) is dated [9 June 2024 in commit  `9cf3b4d`](https://github.com/lovasoa/SQLpage/commit/9cf3b4d5b069e88f2cad22d3c3ae836abacf8b45).

How long does it usually take for SQLpage to catch up with upstream SQLite? What about if a vulnerability is discovered in SQLite?

The risk is to end up using a new SQLite functionality that will not be present in SQLpage's SQLite or, worse, running an insecure version.

**Comment by lovasoa** [comment]
Hi @lyderic !
  Yes, we are dependent on the rust SQLite library we use, and can be a few versions behind on the latest version from upstream. 

  About security: I can recommend this good and detailed overview of security vulnerabilities in SQLite: https://www.sqlite.org/cves.html
  In short: we could run a 10 year old SQLite, and there still wouldn't be any vulnerability reachable by your application users (as opposed to vulnerabilities that you could voluntarily trigger yourself by writing malicious SQL). Of course we cannot promise that it will never happen in the future (in which case we would make an emergency release), but I think sqlite's track record speaks for itself.

**Comment by lyderic** [comment]
Thanks a lot for this detailed answer.
L.

**Comment by lovasoa** [comment, marked as answer]
I opened a ticket upstream: https://github.com/rusqlite/rusqlite/issues/1513

**Comment by lyderic** [comment]
That's great. Thanks!

**Comment by lovasoa** [comment]
Just merged the latest sqlite :)

**Comment by lyderic** [comment]
You rock ;-)
